sql/table_cache.h

```diff
@@ -157,6 +175,20 @@ class Table_cache {
 
   void free_all_unused_tables();
 
+  /**
+    Notify the table cache that for one of its TABLE objects we have
+    finalized loading and parsing triggers.
+  */
+  void notify_triggers_load() {
+    // QQ: Study performance overhead from this. Perhaps it is better to
+    // use atomics instead?
+    lock();
+    m_table_triggers_count++;
```

Comment:
Well... I suspect that realistically there should not be that much difference. As this is not supposed to be hot code path - i.e. cases when you need to load triggers for TABLE object should be normally rare and trigger loading is going to be fairly expensive in any case.
Also it might be more convenient to use the lock()/unlock() in this case if we will decide that we want to be more active enforcing soft limit on TABLE objects with triggers loaded and that we want to check and expel unused objects if number of such objects is exceeded when "opening" table/loading triggers. Current version of thepatch does this only when table is "closed".

Reply:
Implemented.